components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -477,6 +479,12 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
                             errorMessage);
                 }
 
+                String idpName = context.getExternalIdP().getIdPName();
+                String sidClaim = (String) jsonObject.get(OIDCAuthenticatorConstants.SID);
+                if (StringUtils.isNotBlank(sidClaim) && StringUtils.isNotBlank(idpName)) {
+                    context.setProperty(FEDERATED_IDP_SESSION_ID + idpName, sidClaim);
```

Comment:
Let's add a comment explaining why we are adding this here?

Reply:
Added in the 84d75bb2a1e10818cb7c37f084787a25267cff11

Reply:
Also shouldn't this be set as a parameter not as a property?